From dfc731a71afd7c1d13bf1b991b0b8081dcae438d Mon Sep 17 00:00:00 2001
From: PJ Gray <pj@evertrue.com>
Date: Wed, 5 Dec 2012 11:57:30 -0500
Subject: [PATCH] open actionsheet from actionbutton

---
 TSMiniWebBrowser/TSMiniWebBrowser.m | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/TSMiniWebBrowser/TSMiniWebBrowser.m b/TSMiniWebBrowser/TSMiniWebBrowser.m
index 2024ba6..9d68c75 100644
--- a/TSMiniWebBrowser/TSMiniWebBrowser.m
+++ b/TSMiniWebBrowser/TSMiniWebBrowser.m
@@ -346,6 +346,10 @@ - (void)showActionSheet {
     else if (mode == TSMiniWebBrowserModeNavigation && self.navigationController.tabBarController != nil) {
         [actionSheet showFromTabBar:self.navigationController.tabBarController.tabBar];
     }
+    else if (showActionButton && (mode == TSMiniWebBrowserModeModal)) {
+        UIBarButtonItem* actionButton = [[toolBar items] lastObject];
+        [actionSheet showFromBarButtonItem:actionButton animated:YES];
+    }
     else {
         [actionSheet showInView:self.view];
     }
-- 
1.8.4

